lightning/src/ln/channelmanager_limits_tests.rs

```diff
@@ -673,8 +673,8 @@ fn test_channel_update_cached() {
 
 	{
 		// Assert that ChannelUpdate message has been added to node[0] pending broadcast messages
-		let pending_broadcast_messages= nodes[0].node.pending_broadcast_messages.lock().unwrap();
-		assert_eq!(pending_broadcast_messages.len(), 1);
+		// let pending_broadcast_messages= nodes[0].node.get_and_clear_pending_msg_events();
+		// assert_eq!(pending_broadcast_messages.len(), 1);
```

Comment:
Still figuring out how to rewrite this test

Reply:
It looks like if you added a test-only getter that clones the `pending_broadcast_messages`, that would fix it. I don't immediately see a way to replicate this part of the test using the public API (since `get_and_clear`.. clears, which breaks later parts of the test).

Reply:
> It looks like if you added a test-only getter that clones the `pending_broadcast_messages`, that would fix it. I don't immediately see a way to replicate this part of the test using the public API (since `get_and_clear`.. clears, which breaks later parts of the test).

Thanks, I was stuck on this test for a while